### PR TITLE
Reset Keycard functionality

### LINKED ISSUE #9229: Clean keycard and remove multi-account
**Situation** 

Currently, after a keycard has been set-up after onboarding the keycard has 
- setup a PIN, Pairing code, PUK
- secrets loaded (which correspond to status multiaccount seed)

**Problem** 

There is no possibility for the user to remove the secrets loaded physically from the card

A user might want to delete the secrets from his card because
- despite the high level of hardware security of his card, he wants to remove them physically from there to prevent any attack to succeed now or later 
- he wants to change the multiaccount on his card

**Solution** 

A simple approach would be 
- user is logged in his keycard multiaccount
- user goes to 'settings/keycard'
- user choses 'reset keycard' (to be added, does not exist today)
- user is prompted 
"You're about to delete the keys on this card, your pairing code/PUK & PIN remain will unchanged.
You must have your seed if you want to recover any funds from this key in the future. You can now restore another key on this card. Your Pairing code remains unchanged and will be needed if you use a new phone. 
Are you sure you want to proceed ? yes/no"
- if the user clicks 'yes' then 
            - we leaves PIN, PUK, Pairing uncode untouched on the card 
            - the secret on the card is deleted
            - the multiaccount is removed from the phone

### PR DESCRIPTION
**Pull request by bitgamma**
Implements Keycard account removal with (optional) card reset.

Closes #9229 

Screenshot with highlighted new strings for localization

keycard-reset-string-screenshot

status: ready

### CODE REVIEW

File: src/status_im/keycard/delete_key.cljs [hunk 1]
```diff
@@ -74,6 +39,6 @@
                                                  :puk         []
                                                  :status      nil
                                                  :error-label nil
-                                                 :on-verified :keycard/remove-key-with-unpair})}
+                                                 :on-verified (if keep-keys-on-keycard? :keycard/unpair-and-delete :keycard/remove-key-with-unpair)})}
```
Comment (rasom) [inline src/status_im/keycard/delete_key.cljs]: Could you please make it three lines expression? It won't fit 80 chars anyway but this way it is too long

**Review by rasom — APPROVED** [review]

File: src/status_im/keycard/delete_key.cljs (hunk 1 above)
Comment (bitgamma) [reply src/status_im/keycard/delete_key.cljs]: Done. Thanks for the review!

**Comment by qoqobolo** [comment]
@bitgamma Thank you for the great work! Testing is still ongoing.
This issue has been found so far.

### ISSUE 1: No message "2 attempts left" when entering invalid passcode (PIN) during profile deleting. Android only

#### Reproduction

1. Log in to keycard multiaccount on Android device
2. Navigate to Privacy and security settings > Delete my profile
3. Tap `Delete profile`
4. Enter invalid PIN
5. Repeat two more times until the keycard is frozen

#### Expected behavior

Message about how many attempts are left for each input of an incorrect PIN ("2 attempts left", "1 attempt left")

#### Actual behavior

No messages. The PIN screen is closed just after entering an invalid PIN and tapping the keycard

https://user-images.githubusercontent.com/67952253/109811863-a181ac00-7c2b-11eb-944d-5a072f606121.mp4

**Comment by bitgamma** [comment]
@qoqobolo Thanks for the report! Is there any particular prerequisite to reproduce this? I tried on a OnePlus 6 and Xiaomi Redmi Note 8T (both Android 10), and I cannot reproduce it, the number of remaining attempts is shown correctly.

**Comment by qoqobolo** [comment]
> @qoqobolo Thanks for the report! Is there any particular prerequisite to reproduce this? I tried on a OnePlus 6 and Xiaomi Redmi Note 8T (both Android 10), and I cannot reproduce it, the number of remaining attempts is shown correctly.

@bitgamma hmm.. Well, now I can't reproduce it too. My test device is Google Pixel 2 (Android 11).
I'll update the steps as soon as I find them.

**Comment by qoqobolo** [comment]
### ISSUE 2/QUESTION: Can't use keycard after reset. 

@bitgamma 
Also, could you please clarify how exactly keycard is supposed to be used after reset?
Asking, because from the title of the feature request https://github.com/status-im/status-react/issues/9229 and the description it follows that I can use it as an empty card.

But when I try to generate a new multiaccount after reset and choose my card as key storage, after getting a new seed phrase, I am asked to confirm my passcode and after I get this message:
`"Wrong card. Tapped card doesn't correspond to the keys you selected".`

**_Steps:_**
1. Delete keycard-multiaccount
2. Generate keys
3. Choose keycard as key storage
4. Confirm seed phrase

https://user-images.githubusercontent.com/67952253/109837518-8de33f00-7c45-11eb-99ea-83d910ef22f7.mp4

Also, I get the same screen and pop-up if I try to restore my multi-account with seed phrase and chose key storage on my keycard. 

**_Steps:_**
1. Delete keycard-multiaccount
2. Access existing keys
3. Choose keycard as a key storage

https://user-images.githubusercontent.com/67952253/109837879-eadef500-7c45-11eb-8e6f-e1f20ba72969.mp4

If I go through the third flow and try to recover my multiaccount with keycard I get this screen
Screenshot 2021-03-03 at 17 38 13

Could you clarify if this behavior is expected (the first two cases)?
Did I understand correctly that it is assumed that the card can be used for a _**new**_ multi-account with a **_new_** seed phrase after reset?

**Comment by bitgamma** [comment]
@qoqobolo Thanks a lot for catching this! Your assumption are absolutely correct, you should be able to load a new account or enter an existing seed. If you restart Status it works, but if you try to create a new account immediately after deleting it doesn't. I will fix this and ping you.

**Comment by bitgamma** [comment]
@qoqobolo issue 2 is fixed now

**Comment by qoqobolo** [comment]
@bitgamma could you rebase this to the latest develop, please?

**Comment by bitgamma** [comment]
> @bitgamma could you rebase this to the latest develop, please?

@qoqobolo done

**Comment by qoqobolo** [comment]
@bitgamma thanks for the great work!
Failed tests are not related to PR, it's ready to be merged.

**Pull request merged**